rl4co/utils/decoding.py

```diff
@@ -418,6 +423,8 @@ class BeamSearch(DecodingStrategy):
     name = "beam_search"
 
     def __init__(self, beam_width=None, select_best=True, **kwargs) -> None:
+        # TODO do we really need all logp in beam search?
+        kwargs["only_store_selected_logp"] = False
```

Comment:
It looks like its needed in beam search, right? I'm no expert in this, though

Reply:
Currently all logps are used but I am not certain whether this is indeed necessary. I will debug through beam search after the submission

Reply:
All right, yes this would be minor since it is not really used during training